### PR TITLE
Subtitles integration followup

### LINKED ISSUE #3176: Subtitles integration followup
Followup to #3169 

- Allow downloading already published subtitle (via "Download file" context menu option)
- Improve subtitle validation in VideoForm:
    - Disallow publishing without selecting a subtitle file
    - ~Disallow selecting subtitle language if subtitles for that language have already been published~
- Verify that code removed from `UploadStatusGroup.tsx` wasn't needed. Confirm that you can still re-upload file from a new browser. (publish, fail upload, visit app from incognito, verify that the video shows up in failed uploads and you can re-upload)

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #3176

### CODE REVIEW

File: packages/atlas/src/components/_inputs/SubtitlesBox/SubtitlesBox.tsx [hunk 1]
```diff
@@ -97,7 +111,7 @@ export const SubtitlesBox: FC<SubtitleBoxProps> = ({
           {languageIso} {type === 'closed-captions' ? '(CC)' : ''}
         </Text>
         <SubtitlesFileName variant="t100" as="p" color="colorText">
-          {!hasFile ? 'Add subtitles file' : file ? file.name : 'subs.vtt'}
+          {!hasFile ? 'Add subtitles file' : file ? file.name : fileName}
```
Comment (kdembler) [inline packages/atlas/src/components/_inputs/SubtitlesBox/SubtitlesBox.tsx]: So I asked designers about what should be displayed here for published subtitles: https://joystreamworkspace.slack.com/archives/C039GLGPUQ7/p1661857714024849

They suggested we leave this empty in that case, could you update?

Comment (drillprop) [reply packages/atlas/src/components/_inputs/SubtitlesBox/SubtitlesBox.tsx]: @kdembler ok, but should we display the filename once someone uploads a subtitles? 
image
I think we should have the same behavior - don't show filename at all, right now it's confusing to me.

Comment (kdembler) [reply packages/atlas/src/components/_inputs/SubtitlesBox/SubtitlesBox.tsx]: Hmm, that's a valid point. Can you tag Kuba on #atlas-design so we can discuss?

**Review by drillprop — APPROVED** [review]
LGTM

File: packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.tsx [hunk 2]
```diff
@@ -750,10 +747,17 @@ export const VideoForm: FC<VideoFormProps> = memo(({ onSubmit, setFormStatus })
           <Controller
             control={control}
             name="subtitlesArray"
+            rules={{
+              validate: (value) => {
+                const languageWithoutFile = value?.find((language) => !language.file)
+                return !isEdit && value && languageWithoutFile ? 'All selected languages must include a file!' : true
```
Comment (kdembler) [inline packages/atlas/src/views/studio/VideoWorkspace/VideoForm/VideoForm.tsx]: Two things here:
1) Why `!isEdit` as a condition for error? I can still add a new subs language and not provide a file when editing
2) Let's change the message to "Provide a file for every new subtitles language."

**Review by kdembler — CHANGES_REQUESTED** [review]

**Comment by rafalpawlow** [comment]
@kdembler FYI I just added disabling subtitles for minted videos

**Review by kdembler — APPROVED** [review]

**Pull request merged**